dapp/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM node:12.13.0
+
+WORKDIR /app
+
+# cache dependencies in layer
+COPY package.json /app/package.json
+COPY package-lock.json /app/package-lock.json
+RUN npm install
+
+COPY . /app
+RUN npm run build
+
+VOLUME /export
+
+ENTRYPOINT rm -rf /export/dai-hrd; cp -r -p /app/app /export/dai-hrd
```

Comment:
Interesting. feels more like a build step? Or is it for easy debugging?

Reply:
This is a deploy step.  It makes it easy to deploy to containerized IPFS by having this.  Your IPFS container has a shared volume `ipfs-export` that you write into, then you can `docker container exec` on the `ipfs` container after writing into that volume to pin the files.

It allows me to essentially "distribute" the app via docker, and quickly copy it into any IPFS container that has an exposed volume.